### ISSUE TITLE
swr_domain_v2: Forward slashes ("/") in repository names are not handled correctly

### ISSUE THREAD

**Issue by qa-simon-heuser** [opentelekomcloud/terraform-provider-opentelekomcloud#2043]
# Terraform provider version
```
registry.terraform.io/opentelekomcloud/opentelekomcloud v1.32.2
```

# Affected Resource(s)
- opentelekomcloud_swr_domain_v2

# Terraform Configuration Files
```hcl
resource "opentelekomcloud_swr_organization_v2" "test" {
  name = "test"
}

resource "opentelekomcloud_swr_repository_v2" "test" {
  name         = "grafana/grafana"
  organization = opentelekomcloud_swr_organization_v2.test.name
  is_public    = false
  category     = "linux"
}

resource "opentelekomcloud_swr_domain_v2" "sharing" {
  organization  = opentelekomcloud_swr_organization_v2.test.name
  repository    = opentelekomcloud_swr_repository_v2.test.name
  access_domain = "XYZ"
  permission    = "read"
  deadline      = "forever"
}
```

# Debug Output/Panic Output
The error output looks something like this:
```
╷
│ Error: error creating domain: Resource not found: [POST https://swr-api.<region>.otc.t-systems.com/v2/manage/namespaces/test/repos/grafana/grafana/access-domains], error message: {"error_msg":"The API does not exist or has not been published in the environment","error_code":"APIGW.0101","request_id": [...]}
```

# Steps to Reproduce
1. `terraform apply`

# Expected Behavior
The provider should have created the domain sharing by calling the following API endpoint: `https://swr-api.<region>.otc.t-systems.com/v2/manage/namespaces/test/repos/grafana$grafana/access-domains` 
(Note the **$** used to denote slashes in the repo name).

# Actual Behavior
The provider calls the API endpoint described in the error above. Since the slashes are not replaced by `$`s, the request fails.

# Important Factoids
- The correct API endpoint -- in our example, `.../grafana$grafana/access-domains` -- was discovered by analyzing the actual request made by the OTC Console when sharing an image.

# References
- This issue was discovered after #2037 was fixed.

**Comment by anton-sidelnikov** [comment]
@agileman-droid in what region it's happened?

**Comment by qa-simon-heuser** [comment]
@anton-sidelnikov I tested it on `eu-de`

**Comment by anton-sidelnikov** [comment]
Fix available in v1.32.3
